doc/source/parameters/cfd/linear_solver_control.rst

```diff
@@ -0,0 +1,151 @@
+	  set method		= gmres
+
+	  # State whether information from the linear solver should be printed
+	  set verbosity		= verbose
+
+	  # Linear solver minimum residual
+	  set minimum residual  = 1e-12
+
+	  # Linear solver residual
+	  set relative residual = 1e-3
+
+	  # Maximum solver iterations
+	  set max iters         = 1000
+
+	  # Force the linear solver to continue even if it fails
+	  set force linear solver continuation = false
+
+	  # Maximum number of krylov vectors for GMRES and AMG solvers
+	  set max krylov vectors = 100
+
+	  #---------------------------------------------------
+	  # ILU preconditioner parameters for GMRES and BICGSTAB solvers
+	  #---------------------------------------------------
+	  # ILU preconditioner fill
+	  set ilu preconditioner fill = 0
+
+	  # ILU preconditioner tolerance
+	  set ilu preconditioner absolute tolerance = 1e-12
+
+	  # ILU relative tolerance
+	  set ilu preconditioner relative tolerance = 1.00
+
+	  #---------------------------------------------------
+	  # ILU smoother/coarsener parameters for AMG preconditioner
+	  #---------------------------------------------------
+	  # AMG preconditioner ILU smoother/coarsener fill
+	  set amg preconditioner ilu fill = 0
+
+	  # AMG preconditioner ILU smoother/coarsener absolute tolerance
+	  set amg preconditioner ilu absolute tolerance = 1e-12
+
+	  # AMG preconditioner ILU smoother/coarsener relative tolerance
+	  set amg preconditioner ilu relative tolerance = 1.00
+
+	  #---------------------------------------------------
+	  # other AMG solver parameters
+	  #---------------------------------------------------
+	  # AMG aggregation threshold
+	  set amg aggregation threshold = 1e-14
+
+	  # AMG number of cycles
+	  set amg n cycles              = 1
+
+	  # AMG w cycling. If this is set to true, W cycling is used. Otherwise, V cycling is used.
+	  set amg w cycles              = false
+
+	  # AMG smoother sweeps
+	  set amg smoother sweeps       = 2
+
+	  # amg smoother overlap
+	  set amg smoother overlap      = 1
+	end
+
+
+* The ``method`` parameter enables to choose an iterative solver for the linear system of equations. Lethe currently supports these core solution strategies:
+	* ``gmres`` (default parameter value), a GMRES iterative solver with ILU preconditioning.
+	* ``amg``, a GMRES iterative solver with AMG preconditioning and an ILU coarsener and smoother.
+	* ``bicgstab``, a BICGSTAB iterative solver with ILU preconditioning.
+	* ``direct``, a direct solver using `TrilinosWrappers::SolverDirect <https://www.dealii.org/current/doxygen/deal.II/classTrilinosWrappers_1_1SolverDirect.html>`_. 
+
+	.. tip:: 
+		On coarse meshes, the ``gmres``/``bicgstab`` solver with ILU preconditioning is more efficient. 
+
+		As the number of mesh elements increase, the ``amg`` solver is the most efficient. Generally, at 1M elements, the ``amg`` solver always outperforms the ``gmres`` or ``bicgstab`` for 2D steady-state problems.
```

Comment:
```suggestion
		As the number of mesh elements increase, the ``amg`` solver is the most efficient. Generally, at 1M elements, the ``amg`` solver generally outperforms the ``gmres`` or ``bicgstab`` for 2D and 3D steady-state problems. For transient problem, this is much more problem dependent. When the CFL is relatively low, it is generally preferable to use the ``gmres`` solver with a low fill level (e.g. 0) even when the mesh is very large (>10M cells). This is because the transient version of the system of equation is much easier to solve. At large CFL conditions (>10) and/or very large mesh, it may become preferable to transition to the ``amg`` solver. Be aware that the setup of the AMG preconditioner is very expensive and does not scale linearly with the size of the matrix. As such, it is generally preferable to minimize the number of assembly of such preconditioner. This can be achieved by using the ``inexact newton``` (see :doc:`non-linear_solver_control`)
```